src/commands/build.ts

```diff
@@ -116,50 +156,27 @@ export class BuildCommand implements Command {
     logger.info(`Clearing ${mainBuildDirectoryName}${path.sep} directory...`);
     await del([mainBuildDirectoryName]);
 
+    const polymerProject = new PolymerProject(config);
+
     // If any the build command flags were passed as CLI arguments, generate
     // a single build based on those flags alone.
     const hasCliArgumentsPassed =
         this.args.some((arg) => typeof options[arg.name] !== 'undefined');
     if (hasCliArgumentsPassed) {
-      return build(
-          {
-            addServiceWorker: !!options['add-service-worker'],
-            addPushManifest: !!options['add-push-manifest'],
-            swPrecacheConfig: options['sw-precache-config'],
-            insertPrefetchLinks: !!options['insert-prefetch-links'],
-            bundle: !!options['bundle'],
-            html: {
-              minify: !!options['html-minify'],
-            },
-            css: {
-              minify: !!options['css-minify'],
-            },
-            js: {
-              minify: !!options['js-minify'],
-              compile: !!options['js-compile'],
-            },
-          },
-          config);
-    }
-
-    // If no build configurations were passed via CLI flags or the polymer.json
-    // file, generate a default build.
-    if (!config.builds) {
-      return build({}, config);
+      return build(commandOptionsToBuildOptions(options), polymerProject);
     }
 
-    // If a single build was defined or configured via the project config,
-    // generate a build for that configuration.
-    if (config.builds.length === 1) {
-      return build(config.builds[0], config);
+    // If no build flags were passed but 1+ polymer.json build configuration(s)
+    // exist, generate a build for each configuration found.
+    if (config.builds) {
+      return Promise
+          .all(config.builds.map((buildOptions: ProjectBuildOptions) => {
```

Comment:
I don't think you need the argument type, it should be inferred from the type of `config.builds`. You might get down to:

```ts
await Promise.all(config.builds.map((options) => build(options, polymerProject));
```